applications/main/nfc/helpers/protocol_support/slix/slix.c

```diff
@@ -0,0 +1,140 @@
+#include "slix.h"
+#include "slix_render.h"
+
+#include <nfc/protocols/slix/slix_poller.h>
+#include <nfc/protocols/slix/slix_listener.h>
+
+#include "nfc/nfc_app_i.h"
+
+#include "../nfc_protocol_support_gui_common.h"
+
+static void nfc_scene_info_on_enter_slix(NfcApp* instance) {
+    const NfcDevice* device = instance->nfc_device;
+    const SlixData* data = nfc_device_get_data(device, NfcProtocolSlix);
+
+    FuriString* temp_str = furi_string_alloc();
+    furi_string_cat_printf(
+        temp_str, "\e#%s\n", nfc_device_get_name(device, NfcDeviceNameTypeFull));
+    nfc_render_slix_info(data, NfcProtocolFormatTypeFull, temp_str);
+
+    widget_add_text_scroll_element(
+        instance->widget, 0, 0, 128, 64, furi_string_get_cstr(temp_str));
+
+    furi_string_free(temp_str);
+}
+
+static NfcCommand nfc_scene_read_poller_callback_slix(NfcGenericEvent event, void* context) {
+    furi_assert(event.protocol == NfcProtocolSlix);
+
+    NfcApp* instance = context;
+    const SlixPollerEvent* slix_event = event.event_data;
+
+    if(slix_event->type == SlixPollerEventTypeReady) {
+        nfc_device_set_data(
+            instance->nfc_device, NfcProtocolSlix, nfc_poller_get_data(instance->poller));
+        view_dispatcher_send_custom_event(instance->view_dispatcher, NfcCustomEventPollerSuccess);
+        return NfcCommandStop;
+    }
+
+    return NfcCommandContinue;
+}
+
+static void nfc_scene_read_on_enter_slix(NfcApp* instance) {
+    nfc_poller_start(instance->poller, nfc_scene_read_poller_callback_slix, instance);
+}
+
+static void nfc_scene_read_success_on_enter_slix(NfcApp* instance) {
+    const NfcDevice* device = instance->nfc_device;
+    const SlixData* data = nfc_device_get_data(device, NfcProtocolSlix);
+
+    FuriString* temp_str = furi_string_alloc();
+    furi_string_cat_printf(
+        temp_str, "\e#%s\n", nfc_device_get_name(device, NfcDeviceNameTypeFull));
+    nfc_render_slix_info(data, NfcProtocolFormatTypeShort, temp_str);
+
+    widget_add_text_scroll_element(
+        instance->widget, 0, 0, 128, 52, furi_string_get_cstr(temp_str));
+
+    furi_string_free(temp_str);
+}
+
+static NfcCommand nfc_scene_emulate_listener_callback_slix(NfcGenericEvent event, void* context) {
+    furi_assert(context);
+    furi_assert(event.protocol == NfcProtocolSlix);
+    furi_assert(event.event_data);
```

Comment:
In some listeners event_data isn't checked before use. I'm not sure if it really mandatory to check

Reply:
It's a good idea to call `furi_assert()` on all pointer parameters that will be dereferenced directly in the function. If it's just passed along to another function call, then it's better to check it there.